### PR TITLE
Fix unexpected closing of longer modal when clicking on scrollbar

### LINKED ISSUE #2013: Longer Modal closes unexpectedly when interacting with the scrollbar
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Any related issues?

#1909

### Tell us about your environment

Windows 10

### MarkBind version

4.0.0

### What did you do? Describe the bug

After the dual-modal issue is fixed, there's another problem of auto closing modal after an interaction with the scrollbar. This is less obvious as users can scroll via the mouse scroll wheel, which works fine.

### Steps to reproduce

- Open a modal with content longer than a page, which will render the scroll bar
- Click or drag the scrollbar
- See that the modal auto closes

Examples described below,

### Expected behavior

Seeing from one of the previous deployment previews, users can  use the scrollbar
https://deploy-preview-1795--markbind-master.netlify.app/userguide/syntaxcheatsheet
w87Sl6rPX9

### Actual behavior

https://deploy-preview-2012--markbind-master.netlify.app/userguide/syntaxcheatsheet
Modal auto closes once user releases his/her click on the scrollbar
mqQoIkVh81

the 

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by WillCWX**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [X] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

Fixes #2013 
Fixes this of #1916 :
> When the modal is triggered, the scroll bar appears to come about from the center of the screen and shifts immediately to the right, a little weird to see that.

**Overview of changes:**

- Moves scroll bar to `div class="modal-body"`.
- Other CSS adjusted to keep the gap between the dialog window and the browse window
- Updates `modal` tests accordingly.

**Anything you'd like to highlight/discuss:**

The `max-height` of the `div class="modal-content"` is set to `90vh` to enable scrolling as `max-height` is infinite otherwise.
`90vh` looks best to me but I don't have very good eyes for design so do play around with the number to see what maybe best.

I have not tested scrolling in the `x-axis` as I have yet to find a `modal` that is longer than the webpage but adding `max-width=90vw` may fix this easily as well. 

**Testing instructions:**

Add this [topic1.md](https://github.com/MarkBind/markbind/files/11969609/topic1.md) file to a markbind website.
The `modal` should have a scrollbar that is clickable and looks like this:

image
_the white bar is from snip and not from the website_

**Proposed commit message: (wrap lines at 72 characters)**
Fix unexpected closing of longer modal when clicking on scrollbar

Fix modal scroll
---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [X] Linked all related issues
- [X] No unrelated changes

### CODE REVIEW

File: packages/vue-components/src/Modal.vue [hunk 1]
```diff
@@ -131,6 +130,10 @@ export default {
         overflow: auto;
     }
 
+    .max-height {
+        max-height: 90vh;
```
Comment (yucheng11122017) [inline packages/vue-components/src/Modal.vue:134]: Personally I think its better to have the max height as 100vh to be consistent with previous display

File: packages/vue-components/src/Modal.vue [hunk 2]
```diff
@@ -6,15 +6,14 @@
...
-    classes="allow-overflow"
     :content-class="['modal-dialog', optionalModalSize, optionalCentering]"
     overlay-transition="none"
     :transition="effectClass"
     :click-to-close="backdrop !== 'false'"
     esc-to-close
     z-index-base="2000"
   >
-    <div class="modal-content">
+    <div class="modal-content allow-overflow max-height">
```
[the remark this answers is not in the document]
Comment (WillCWX) [reply packages/vue-components/src/Modal.vue:16]: Do we keep `allow-overflow` as well or merge it into `modal-content`?

File: packages/vue-components/src/Modal.vue (hunk 1 above)
Comment (WillCWX) [reply packages/vue-components/src/Modal.vue:134]: The thing is that when you scroll to the bottom, there is this nice gap for the previous display
image

But will be absent with `100vh`
image

Will that still be ok?

Comment (yucheng11122017) [reply packages/vue-components/src/Modal.vue:134]: oh I understand where you are coming from. Ill still prefer the 100vh though, I think the gap below looks a little awkward with 90

File: packages/vue-components/src/Modal.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (yucheng11122017) [reply packages/vue-components/src/Modal.vue:16]: Keep it would be good :)

[the remark this answers is not in the document]
Comment (WillCWX) [reply packages/vue-components/src/Modal.vue:16]: Sorry, I didn't realize but `modal-content` is from `bootstrap.min.css` file in `core-web/asset/css` rather then defined by us. I'm not sure whether I can change stuff in that file.
Is it ok for me to tuck `max-height` under `allow-overflow` instead?

[the remark this answers is not in the document]
Comment (yucheng11122017) [reply packages/vue-components/src/Modal.vue:16]: I believe you can add the max-height under modal-content by defining it in the style tag in the vue component as so:

```
<style>
.modal-content {
   max-height: 100vh
}
...
</style
```

**Review by yucheng11122017 — APPROVED** [review]
LGTM! Thanks for the work @WillCWX

**Comment by tlylt** [comment]
Hi @WillCWX, wondering if you have explored ways to make it work without moving the scrollbar into the modal? Would be great if we could provide both options, perhaps something like what Chakra UI offers:
> 
image

https://chakra-ui.com/docs/components/modal/usage#modal-overflow-behavior

**Comment by tlylt** [comment]
> Personally I think its better to have the max height as 100vh to be consistent with previous display

@yucheng11122017 what do you mean by previous display? Existing modal without this PR change has a "padding" at the bottom, which is achievable if we do with max height = 100% + some other CSS. I think this might be more comfortable visually hence let's go with this instead 😺.
image

I adjusted the style slightly to
- achieve the margin around the dialog window
- move the scrollbar into the modal body instead of having it scroll from the header
- and also updated the test case.

This is not the best solution out there but I think it does address the issue for now. In fact, scrolling within the body seems to be the solution for long modals for [v2 of vue-final modal](https://v2.vue-final-modal.org/examples/livedemo/) (slight difference in that they have more margin) :
image


Will post a separate issue to track modal improvements.

Some investigation note:
- the current issue #2013 has been reported [here](https://github.com/vue-final/vue-final-modal/issues/337) in vue-final-modal
- an example is given by the author [here](https://codesandbox.io/p/sandbox/vue-final-modal-scroll-forked-totdfc?file=%2Fsrc%2Fcomponents%2FModalConfirm.vue%3A36%2C1-41%2C2) and [here](https://vue-final-modal.org/use-cases/modal-long-scroll), however these examples are using later version of vue-final-modal. I tried to implement something similar but could not get it to work

**Review by tlylt — APPROVED** [review]

**Comment by tlylt** [comment]
@all-contributors please add @WillCWX for code

**Pull request merged**